client/client.go

```diff
@@ -335,6 +336,18 @@ func (c *Client) processChannelRequests() {
 	c.l.Debug("Channel closed.")
 }
 
+func (c *Client) getActionTimeout(req *agentpb.StartActionRequest) time.Duration {
+	d, err := ptypes.Duration(req.Timeout)
+	if d == 0 {
```

Comment:
it always 0 if `err != nil`, it means that `err` will be overridden by the block inside.

Reply:
Yes, and we even have a test for that

Reply:
Ah, got it